filecoin-project/lotus
# tests: itests: test create2 and destroy

## Related Issues
 https://github.com/filecoin-project/ref-fvm/issues/1491

## Proposed Changes

- Create2 factory contract
- use factory to create2 destroyable contract
- test stored address property "sender"
- run test() which 1.calls selfdestruct 2. verifies sender() is the correct value 3. attempts and fails to deploy via create2
- test stored property (fails)
- deploy new contract at same address using same salt
- verify sender() property is correct

## Additional Info
this test is currently blocked on fvm updates

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
